lib/hooks.ts

```diff
@@ -19,15 +31,38 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
+export function useProductList(query?: QueryParams) {
     const { context } = useSession();
+    const params = new URLSearchParams({ ...query, context }).toString();
+
     // Use an array to send multiple arguments to fetcher
-    const { data, error, mutate: mutateList } = useSWR(context ? ['/api/products/list', context] : null, fetcher);
+    const { data, error, mutate: mutateList } = useSWR(context ? ['/api/products/list', params] : null, fetcher);
 
     return {
-        list: data,
+        list: data?.data,
+        meta: data?.meta,
         isLoading: !data && !error,
-        isError: error,
+        error,
         mutateList,
     };
 }
+
+export function useProductInfo(pid: number, list?:ListItem[]) {
+    const { context } = useSession();
+    const params = new URLSearchParams({ context }).toString();
+
+    let product: ListItem; 
+
+    if (list?.length) { 
+       product = list.find(item => item.id === pid);
+    }
+
+    // Conditionally fetch product if it doesn't exist in the list (e.g. deep linking)
+    const { data, error } = useSWR(!product && context ? [`/api/products/${pid}`, params] : null, fetcher);
+
+    return {
+        product: product ?? data,
+        isLoading: product ? false : (!data && !error),
+        error,
+    };
+}
```

Comment:
Note: The main branch contains more hooks for orders, shipping, etc. However I didn't add them in this file, since the tutorial doesn't mention that. Or do we also want to add them as well? Up to us?  Reference: https://github.com/bigcommerce/sample-app-nodejs/blob/main/lib/hooks.ts#L70